appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockLineChartModel.java

```diff
@@ -0,0 +1,38 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import org.pepstock.charba.client.AbstractChart;
+import org.pepstock.charba.client.Chart;
+import org.pepstock.charba.client.data.DataPoint;
+import org.pepstock.charba.client.data.LineDataset;
+
+import java.util.ArrayList;
+
+public class MockLineChartModel extends MockChartModel<LineDataset> {
+
+    public MockLineChartModel() {
+        dataSeries = new LineDataset();
+
+        dataSeries.setFill(false);
+        dataSeries.setBorderWidth(1);
+        dataSeries.setLineTension(0);
+    }
+
+    @Override
+    public void addEntry(float x, float y) {
+        DataPoint point = new DataPoint();
+        point.setX(x);
+        point.setY(y);
+        dataSeries.getDataPoints(true).add(point);
```

Comment:
Possibly a cleaner way then using getDataPoints with true parameter, which always initializes a new List object.

Reply:
Alternatively, an individual add entry method is actually not even needed, because there should not be a case where individual entries can be added on to a Chart.